src/Mbstring/Mbstring.php

```diff
@@ -533,18 +535,20 @@ public static function mb_str_split($string, $split_length = 1, $encoding = null
 
             return null;
         }
-
-        if (1 > $split_length = (int) $split_length) {
-            trigger_error('The length of each segment must be greater than zero', E_USER_WARNING);
-
-            return false;
+        if (null !== $split_length && !\is_bool($split_length) && !\is_numeric($split_length)) {
+            trigger_error('mb_str_split() expects parameter 2 to be int, '.\gettype($split_length).' given', E_USER_WARNING);
+            
+            return null;
         }
+        $split_length = (int) $split_length;
+        if (1 > $split_length) {
+            trigger_error('mb_str_split(): The length of each segment must be greater than zero', E_USER_WARNING);
```

Comment:
can you add tests for the two new warnings please?

Reply:
@nicolas-grekas
~~~php
$split_length = 1.2;
var_dump(1.1 > $split_length = (int) $split_length); // bool(true) 1.1>1
~~~

Reply:
Since this is a polyfill, the correct behavior is easy to decide: how does the native function behave? To verify and prevent regressions, the correct behavior should be added as a test case, if one is missing.

Reply:
@nicolas-grekas need fix ci configs before update tests